pkgs/development/python-modules/argostranslate/default.nix

```diff
@@ -0,0 +1,67 @@
+{ lib
+, buildPythonPackage
+, fetchPypi
+, pytestCheckHook
+, ctranslate2
+, ctranslate2-cpp
+, sentencepiece
+, stanza
+}:
+let
+  ctranslate2OneDNN = ctranslate2.override {
+    ctranslate2-cpp = ctranslate2-cpp.override {
+      # https://github.com/OpenNMT/CTranslate2/issues/1294
+      withOneDNN = true;
+      withOpenblas = false;
+    };
+  };
+in
+buildPythonPackage rec {
+  pname = "argostranslate";
+  version = "1.8.0";
+
+  format = "setuptools";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "9b109255d6a2c692c6f3bfbde494d1a27b3d5ed1c1d1d78711cdc1b1e3744c64";
+  };
+
+  propagatedBuildInputs = [
+    ctranslate2OneDNN
+    sentencepiece
+    stanza
+  ];
+
+  postPatch = ''
+    ln -s */requires.txt requirements.txt
+
+    substituteInPlace requirements.txt  \
+      --replace "==" ">="
+  '';
+
+  doCheck = false; # needs network access
+
+  nativeCheckInputs = [
+    pytestCheckHook
+  ];
+
+  # required for import check to work
+  # PermissionError: [Errno 13] Permission denied: '/homeless-shelter'
```

Comment:
Likely means you need to fix up `HOME`.

Reply:
What do you mean? It's just creating a [bunch of directories on import](https://github.com/argosopentech/argos-translate/blob/401ca767db3cf3b6f2db5fb004488fa5b350defb/argostranslate/settings.py) in `~/.local`

Reply:
I mean the homeless-shelter error means, that it wants to access something below `$HOME`. And if that is required for the tests to work it should be set, e.g. to `HOME=$TMPDIR`.

Reply:
Updated!